src/common/Evidence.h

```diff
@@ -147,7 +147,40 @@ class Evidence {
     return std::time(nullptr);
   }
 
-  // given a date time value, stripe time out to date granularity
+  // given a date time value, strip all but year
+  // such as value of (2019-05-15 23:01:34) => value of (2019-01-1 00:00:00)
+  static std::time_t year(std::time_t time) {
+    // remove all except for date
+    auto strip_days = date(time);
+    auto tm = std::gmtime(&strip_days);
+    // yday stores [0-365]
+    auto delta_days = (tm->tm_yday) * 86400; 
+    return strip_days - delta_days;
+  }
+
+  // given a date time value, strip all but month
+  // such as value of (2019-05-15 23:01:34) => value of (2019-05-1 00:00:00)
+  static std::time_t month(std::time_t time) {
```

Comment:
you should write unit tests for all these new functions in CommonTests

Reply:
if you have done this item, could you please reply and close it, so I don't need to track these open comments, we want to eventually close all the comments and then we will merge/

Reply:
done